indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -485,7 +488,8 @@ static Map<Interval, DataSchema> createDataSchemasForIntervals(
       @Nullable final ClientCompactionTaskTransformSpec transformSpec,
       @Nullable final AggregatorFactory[] metricsSpec,
       @Nullable final ClientCompactionTaskGranularitySpec granularitySpec,
-      final ServiceMetricEvent.Builder metricBuilder
+      final ServiceMetricEvent.Builder metricBuilder,
+      CompactionRunner compactionRunner
```

Comment:
I think we should make the `createDataSchema()` and `createDataSchemasForIntervals()` non-static and remove some arguments from the constructor since most of them are already fields of the class.

Reply:
Agree. Will do this in the next PR since it requires updating many tests which use this as a static function without creating a `CompactionTask`.